android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -616,6 +621,57 @@ public void run() {
         }.start();
     }
 
+    /**
+     * Registers a wifi network in the device wireless networks
+     * For API >= 30 uses intent to permanently store such network in user configuration
+     * For API <= 29 uses deprecated functions that manipulate directly
+     * *** registerWifiNetwork :
+     * param ssid, SSID to register
+     * param password, passphrase to use
+     * param security, security mode (WPA or null) to use
+     * return {@code true} if the operation succeeds, {@code false} otherwise
+     */
+    private void registerWifiNetwork(final MethodCall poCall, final Result poResult) {
+        String ssid = poCall.argument("ssid");
+        String password = poCall.argument("password");
+        String security = poCall.argument("security");
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            final WifiNetworkSuggestion.Builder suggestedNet = new WifiNetworkSuggestion.Builder();
+            suggestedNet.setSsid(ssid);
+
+            if (security != null && security.toUpperCase().equals("WPA")) {
+                suggestedNet.setWpa2Passphrase(password);
+            } else if (security != null && security.toUpperCase().equals("WEP")) {
+                // WEP is not supported
+                poResult.error("Error", "WEP is not supported for Android SDK " + Build.VERSION.SDK_INT, "");
```

Comment:
This is missing a `return` statement. Otherwise, this will fall through and call `poResult.success(null);` further down this block.

Reply:
Many thanks @diegotori for catching this! Fixed now found also one more occurrence from previous code (in that case would not return success but keep doing useless operation that would fail otherwise)